### PR TITLE
Dense matmul

### PR DESCRIPTION
**Pull request by Bihaqo**

### CODE REVIEW

File: ops.py [hunk 1]
```diff
@@ -126,7 +126,47 @@ def tt_dense_matmul(tt_matrix_a, matrix_b):
...
+      raise ValueError('Arguments shapes should align got %d and %d instead.' %
+                       (tt_matrix_a.get_shape(), matrix_b.get_shape()))
+
+  if tt_matrix_a.get_shape().is_fully_defined():
+    a_shape = tt_matrix_a.get_shape().as_list()
+    a_raw_shape = [s.as_list() for s in tt_matrix_a.get_raw_shape()]
+  else:
+    a_shape = shapes.shape(tt_matrix_a)
+    a_raw_shape = shapes.raw_shape(tt_matrix_a)
+  if matrix_b.get_shape().is_fully_defined():
```
Comment (geffy) [inline ops.py]: this `if` used three times -- should be a function

File: ops.py [hunk 2]
```diff
@@ -126,7 +126,47 @@ def tt_dense_matmul(tt_matrix_a, matrix_b):
...
+    b_shape = tf.shape(matrix_b)
+  if tt_matrix_a.get_tt_ranks().is_fully_defined():
+    a_ranks = tt_matrix_a.get_tt_ranks().as_list()
+  else:
+    a_ranks = shapes.tt_ranks(tt_matrix_a)
+  # If A is (i0, ..., id-1) x (j0, ..., jd-1) and B is (j0, ..., jd-1) x K,
+  # data is (K, j0, ..., jd-2) x jd-1 x 1
+  data = tf.transpose(matrix_b)
+  data = tf.reshape(data, (-1, a_raw_shape[1][-1], 1))
+  for core_idx in range(ndims - 1, -1, -1):
```
Comment (geffy) [inline ops.py]: `range(ndims - 1, -1, -1)` or `reversed(range(ndims))`?

File: ops.py (hunk 1 above)
Comment (Bihaqo) [inline ops.py]: #16

File: ops.py (hunk 2 above)
Comment (Bihaqo) [inline ops.py]: Fixed

**Pull request merged**